src/consumption.cpp

```diff
@@ -560,6 +560,9 @@ int Character::vitamin_mod( const vitamin_id &vit, int qty )
     } else if( qty < 0 ) {
         it->second = std::max( it->second + qty, v.min() );
         update_vitamins( vit );
+        for( const std::pair<vitamin_id, int> &dcy : v.decays_into() ) {
+            vitamin_mod( dcy.first, dcy.second * ( it->second == 0 ? 0 : std::abs( qty ) ) );
```

Comment:
Can't you just make this a conditional?

Like if (it->second != 0) { do the vitamin mod }



Since moding a vitamin by something * 0 is just modifying it by 0.

Reply:
This is definitely temporary; I mentioned in the OP that I'd be improving it, don't worry. :P